src/third_party/starboard/wpe/shared/drm/drm_system_ocdm.cc

```diff
@@ -430,9 +433,18 @@ void DrmSystemOcdm::GenerateSessionUpdateRequest(
       new Session(this, ocdm_system_, context_,
                   session_update_request_callback_, session_updated_callback_,
                   key_statuses_changed_callback_, session_closed_callback_));
-  session->GenerateChallenge(type, initialization_data,
-                             initialization_data_size, ticket);
-  sessions_.push_back(std::move(session));
+  SbDrmStatus result = session->GenerateChallenge(type, initialization_data, initialization_data_size, ticket);
+  if (result == kSbDrmStatusSuccess) {
+    sessions_.push_back(std::move(session));
+  } else if (result == kSbDrmStatusQuotaExceededError) {
+    // Key rotation use-case: Max limit of 16 concurrent sessions is reached.
+    // Release previously created sessions and attempt again.
+    sessions_.erase(sessions_.begin(), std::prev(sessions_.end()));
+    SbDrmStatus result = session->GenerateChallenge(type, initialization_data, initialization_data_size, ticket);
+    if (result == kSbDrmStatusSuccess) {
```

Comment:
Can there be other failures?

Reply:
Other error is kSbDrmStatusUnknownError when opencdm_construct_session() API fails synchronously. It is notified to upstream application in GenerateChallenge() function itself.